sqlobject/src/main/java/org/jdbi/v3/sqlobject/BridgeMethodHandlerFactory.java

```diff
@@ -36,7 +37,30 @@
                 return IntStream.range(0, method.getParameterCount())
                     .allMatch(i -> methodParamTypes[i].isAssignableFrom(candidateParamTypes[i]));
             })
-            .<Handler>map(m -> (target, args, handle) -> m.invoke(target, args))
+            .<Handler>map(m -> {
+                final MethodHandle mh = unreflect(sqlObjectType, m);
+                return (target, args, handle) -> {
+                    try {
+                        return mh.bindTo(target).invokeWithArguments(args);
+                    } catch (Throwable t) { // Handle et al should <X extends Throwable> not Exception!!!
+                        sneakyThrow(t);
```

Comment:
The Java compiler is horrible here :(

Reply:
Yes.  Also, I think it shows that we erred in our `<X extends Exception>` it should have been `<X extends Throwable>`

Reply:
Is it time to start keeping track of these mistakes in a 4.x wishlist document, the way we did from 2.x to 3.x?